app/experimenter/kinto/tasks/nimbus.py

```diff
@@ -0,0 +1,164 @@
+import markus
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from experimenter.celery import app
+from experimenter.experiments.api.v6.serializers import NimbusExperimentSerializer
+from experimenter.experiments.changelog_utils import generate_nimbus_changelog
+from experimenter.experiments.models import (
+    NimbusBucketRange,
+    NimbusExperiment,
+    NimbusIsolationGroup,
+)
+from experimenter.kinto.client import KintoClient
+
+logger = get_task_logger(__name__)
+metrics = markus.get_metrics("kinto.tasks")
+
+
+def get_kinto_user():
+    user, _ = get_user_model().objects.get_or_create(
+        email=settings.KINTO_DEFAULT_CHANGELOG_USER,
+        username=settings.KINTO_DEFAULT_CHANGELOG_USER,
+    )
+    return user
+
+
+@app.task
+@metrics.timer_decorator("push_experiment_to_kinto.timing")
+def nimbus_push_experiment_to_kinto(experiment_id):
+    kinto_client = KintoClient(settings.KINTO_COLLECTION_NIMBUS)
+
+    metrics.incr("push_experiment_to_kinto.started")
+
+    experiment = NimbusExperiment.objects.get(id=experiment_id)
+    if not NimbusBucketRange.objects.filter(experiment=experiment).exists():
+        NimbusIsolationGroup.request_isolation_group_buckets(
+            experiment.slug,
+            experiment,
+            NimbusExperiment.BUCKET_COUNT,
+        )
+
+    data = NimbusExperimentSerializer(experiment).data
+
+    logger.info(f"Pushing {experiment} to Kinto")
+
+    try:
+        kinto_client.push_to_kinto(data)
+
+        experiment.status = NimbusExperiment.Status.ACCEPTED
+        experiment.save()
+
+        generate_nimbus_changelog(experiment, get_kinto_user())
+
+        logger.info(f"{experiment} pushed to Kinto")
+        metrics.incr("push_experiment_to_kinto.completed")
+    except Exception as e:
+        metrics.incr("push_experiment_to_kinto.failed")
+        logger.info(f"Pushing {experiment} to Kinto failed: {e}")
+        raise e
+
+
+@app.task
+@metrics.timer_decorator("check_kinto_push_queue")
+def nimbus_check_kinto_push_queue():
+    kinto_client = KintoClient(settings.KINTO_COLLECTION_NIMBUS)
+
+    metrics.incr("check_kinto_push_queue.started")
+
+    queued_experiments = NimbusExperiment.objects.filter(
+        status=NimbusExperiment.Status.REVIEW
+    )
+
+    rejected_collection_data = kinto_client.get_rejected_collection_data()
+    if rejected_collection_data:
+        rejected_slug = kinto_client.get_rejected_record()
+        experiment = NimbusExperiment.objects.get(slug=rejected_slug)
+        experiment.status = NimbusExperiment.Status.DRAFT
+        experiment.save()
+
+        generate_nimbus_changelog(
+            experiment,
+            get_kinto_user(),
+            message=f'Rejected: {rejected_collection_data["last_reviewer_comment"]}',
+        )
+
+        kinto_client.delete_rejected_record(rejected_slug)
```

Comment:
It's only possible during rejection for a single record to be rejected at a time? There can't be 3 rejected nimbus experiments?

Reply:
Yeah exactly, a limitation on remote settings is that only a single record can be in review at a time, and there's only a single instance of the task that communicates with remote settings, so it'll always hit this first, unpack the rejected experiment before it pushes another one for review.

Reply:
Ok, so to make sure I understand this right.... Kinto only allows a single record to be in review, so that's why this code only pushes a single record at a time? Does that mean that once a person requests a review, no more records can be pushed until its been reviewed?

Reply:
Yes exactly

Reply:
Great, excellent doc string.